apps/roam/src/utils/exportUtils.ts

```diff
@@ -0,0 +1,141 @@
+import type { Result } from "roamjs-components/types/query-builder";
+import { PullBlock, TreeNode, ViewType } from "roamjs-components/types";
+import type { DiscourseNode } from "./getDiscourseNodes";
+import matchDiscourseNode from "./matchDiscourseNode";
+
+type DiscourseExportResult = Result & { type: string };
+
+export const uniqJsonArray = <T extends Record<string, unknown>>(arr: T[]) =>
+  Array.from(
+    new Set(
+      arr.map((r) =>
+        JSON.stringify(
+          Object.entries(r).sort(([k], [k2]) => k.localeCompare(k2)),
+        ),
+      ),
+    ),
+  ).map((entries) => Object.fromEntries(JSON.parse(entries))) as T[];
```

Comment:
Unsafe argument of type `any` assigned to a parameter of type `Iterable<readonly [PropertyKey, unknown]>`.eslint@typescript-eslint/no-unsafe-argument